pkg/scheduler/plugins/gpushare/device_cache.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gpushare
+
+import (
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	quotav1 "k8s.io/apiserver/pkg/quota/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	apiext "github.com/koordinator-sh/koordinator/apis/extension"
+	schedulingv1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+)
+
+type nodeDeviceCache struct {
+	lock sync.Mutex
+	// nodeDeviceInfos stores nodeDevice for each node
+	// and uses node name as map key.
+	nodeDeviceInfos map[string]*nodeDevice
+}
+
+type nodeDevice struct {
+	lock        sync.Mutex
+	deviceTotal map[schedulingv1alpha1.DeviceType]*deviceResources
+	deviceFree  map[schedulingv1alpha1.DeviceType]*deviceResources
+	deviceUsed  map[schedulingv1alpha1.DeviceType]*deviceResources
```

Comment:
map[schedulingv1alpha1.DeviceType]*deviceResources-> map[schedulingv1alpha1.DeviceType]deviceResources;  map is pointer reference already.